### ISSUE TITLE
IHostingEnvironment IsContainer extension method

### ISSUE THREAD

**Issue by sefacan** [dotnet/extensions#2863]
It can be a good help for container applications other than development mode. The application can be in both development mode and container. Both may be needed. 
The ``DOTNET_RUNNING_IN_CONTAINER = true`` can be used for this.

**Comment by sefacan** [comment]
@davidfowl I'm reading the database connection information from the settings file in the development environment.  I read this information from environment variables when working on Docker.  In the production environment, I specify that it read from environment variables, but I also want to run it on container in development mode.

**Comment by davidfowl** [comment]
Sorry I don't understand why that logic would change if you're in a container. Why aren't you using configuration in both cases? Can you show the code snippet you have today and what you would like to write instead?

**Comment by sefacan** [comment]
As another example, using redis in the container can help to work with memory cache when developing.  I want to know if the application is in both development mode and in container.

**Comment by davidfowl** [comment]
I don’t think that makes sense either. Running in a container shouldn’t change anything. You should be using configuration to determine that not detecting that you’re running in a container

**Comment by anurse** [comment]
This doesn't seem like something that `Microsoft.Extensions.Hosting` should be providing. This feels like it belongs in dotnet/runtime instead, since it's an invariant that's true regardless of if you're in a Hosting-based app.
